pkg/apis/core/types.go

```diff
@@ -157,6 +157,31 @@ type VolumeSource struct {
 	// CSI (Container Storage Interface) represents ephemeral storage that is handled by certain external CSI drivers (Beta feature).
 	// +optional
 	CSI *CSIVolumeSource
+	// Ephemeral represents a volume that is handled by a normal storage driver (Alpha feature).
+	// The volume will be created and deleted together with the pod ("ephemeral volume").
+	//
+	// Use this if:
+	// a) the volume is only needed while the pod runs,
+	// b) features of normal volumes like restoring from snapshot or capacity
+	//    tracking are needed,
+	// c) the storage driver is specified through a storage class, and
+	// d) the storage driver supports dynamic volume provisioning through
+	//    a PersistentVolumeClaim (see EphemeralVolumeSource for more
+	//    information on the connection between this volume type
+	//    and PersistentVolumeClaim).
+	//
+	// Use PersistentVolumeClaim or one of the vendor-specific
+	// APIs for persistent volumes.
+	//
+	// Use CSI for light-weight local ephemeral volumes if the CSI driver is meant to
+	// be used that way - see the documentation of the driver for
+	// more information.
+	//
+	// A pod can use both types of ephemeral volumes and
```

Comment:
> A pod can use both ephemeral and persistent volumes at the same time as long as each volume claim has a unique name.

That actually is another validation thing - is someone allowed to create a PVC mount that references the name that the ephemeral volume would create?  I.e. pod references PVC `foo-bar-baz` and the pod name is `foo-bar` and there is an ephemeral volume `baz` - is that also rejected?

Reply:
It is allowed. Whether it works depends on whether the storage driver supports concurrent access by two different pods. This could lead to problems like one pod starting to use a PVC that was meant to be used by another; that's not different from PVC name conflicts between independent apps that get deployed in the same namespace.

Should we prevent it? I don't think so. The promise is that Kubernetes manages these volumes for the pod, but there is no promise that only one pod gets access. The access boundary is the namespace, just as it is for normal PVCs.

To phrase it differently: that the PVC is visible in the cluster is a feature, not a bug. It enables things like taking a snapshot of the volume (for example, for debugging), resizing it, checking its status, etc.

Reply:
Sorry, i mean - within the same pod.  What use case would be enabled by allowing a PVC claim in a pod and an ephemeral definiton in a pod to use the same target PVC name?  If we can't think of one, disallow by default.

Reply:
Ah, okay. Yes, that can be prevented during validation. Will add.

Reply:
Done and pushed. The error looks like this:
```
volumes[1].persistentVolumeClaim.claimName: Invalid value: "test-pod-ephemeral-volume": must not reference a PVC that gets created for an ephemeral volume
```